### PR TITLE
Enabling WAAPI animations for `opacity`

### PR DESCRIPTION
**Pull request by mattgperry**
Implements https://github.com/framer/motion/issues/1806

### CODE REVIEW

File: packages/framer-motion/src/animation/legacy-popmotion/index.ts [hunk 1]
```diff
@@ -157,12 +158,15 @@ export function animate<V = number>({
         driverControls.start()
     }
 
-    play()
+    autoplay && play()
```
Comment (nvh) [inline packages/framer-motion/src/animation/legacy-popmotion/index.ts:161]: Heh, this was removed in the previous PR.

File: packages/framer-motion/src/animation/legacy-popmotion/index.ts [hunk 2]
```diff
@@ -157,12 +158,15 @@ export function animate<V = number>({
...
 
     return {
         stop: () => {
             onStop && onStop()
             driverControls.stop()
         },
+        sampleForT: (t: number) => {
+            return animation.next(Math.max(0, t)).value
+        },
```
[the remark this answers is not in the document]
Comment (nvh) [reply packages/framer-motion/src/animation/legacy-popmotion/index.ts:170]: Can we add a comment or change the name?

File: packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { EasingDefinition } from "../../easing/types"
+import type { VisualElement } from "../../render/VisualElement"
+import type { MotionValue } from "../../value"
+import { animate } from "../legacy-popmotion"
+import { spring } from "../legacy-popmotion/spring"
+import { AnimationOptions } from "../types"
+import { animateStyle } from "./"
+
+const sampleDelta = 10 //ms
```
Comment (nvh) [inline packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts:9]: Is there any science in picking this constant? Let's add a comment

File: packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        if (currentTime) {
+            const sampleAnimation = animate(options)
+            value.setWithVelocity(
+                sampleAnimation.sampleForT(currentTime - sampleDelta),
+                sampleAnimation.sampleForT(currentTime),
+                sampleDelta
+            )
+        }
+
+        animation.cancel()
```
Comment (nvh) [inline packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts:96]: Is there no `onCancel` callback?

**Review by nvh — APPROVED** [review]
A couple of questions/remarks, but nothing serious.

File: packages/framer-motion/src/animation/legacy-popmotion/index.ts (hunk 1 above)
Comment (mattgperry) [reply packages/framer-motion/src/animation/legacy-popmotion/index.ts:161]: It was sods law that I went from "hey we've never once used this in Motion" to "i don't need to actually play this animation"

File: packages/framer-motion/src/animation/legacy-popmotion/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattgperry) [reply packages/framer-motion/src/animation/legacy-popmotion/index.ts:170]: Sure maybe I'll just call it `sample` - `t` is commonly used in the code for "time" and `p` for "progress"

File: packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts (hunk 3 above)
Comment (mattgperry) [reply packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts:9]: Good point - added

```
/**
 * 10ms is chosen here as it strikes a balance between smooth
 * results (more than one keyframe per frame at 60fps) and
 * keyframe quantity.
 */
```

File: packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts (hunk 4 above)
Comment (mattgperry) [reply packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts:96]: This function is called by the `MotionValue` when a new animation interrupts this one (I'll add a comment) so any cancel callbacks will be handled by that (as of the MotionValue event PR)

**Pull request merged**